### ISSUE TITLE
Some DB tables are not dumped when using `db.to_csv()`

### ISSUE THREAD

**Issue by nesnoj** [OpenEnergyPlatform/open-MaStR#383]
Hey!

It's great to see the progress you've made this year :+1:. Following your [instructions in the docs](https://open-mastr.readthedocs.io/en/latest/getting_started.html) I was able to bulk-download and access the MaStR data. But:

### Expected behavior

`db.to_csv(None)` exports all tables.

### Problem

`db.to_csv(None)` successfully creates a dump but some tables seem to be omitted, e.g. no files are created for locations and gas. So then I tried
```
>>> db.to_csv("location")
are saved to: /home/nesnoj/.open-MaStR/data/dataversion-2022-11-17
```
afterwards, but no location file is created.
So I thought it might be a problem if the dir already exists, removed it and explicitly listed all tables:
```
>>> db.to_csv(['wind', 'solar', 'biomass', 'hydro', 'gsgk', 'combustion', 'nuclear', 'gas', 'storage', 'electricity_consumer', 'location', 'market', 'grid', 'balancing_area', 'permit', 'deleted_units'])

Technology tables: ['wind', 'solar', 'biomass', 'hydro', 'gsgk', 'combustion', 'nuclear', 'storage']

Additional tables: ['electricity_consumer', 'balancing_area']
are saved to: /home/nesnoj/.open-MaStR/data/dataversion-2022-11-17
```
Same result, the additional tables are missing again.

### Investigation

Ok, as the requested table "location" is not included in the "Additional tables: ..." above, I checked these lines in the package to find out where they get lost:

https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/mastr.py#L313-L325

It turns out that it is included in data before the loop starts in L316 but not in the constant `ADDITIONAL_TABLES` taken from [`constants.py`](https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/utils/constants.py#L49-L61) - in there only "locations_extended" is listed. However, `db.to_csv("locations_extended")` is not allowed and raises [this message](https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/utils/helpers.py#L219).

I guess this is a bug?

My dirty workaround: Comment out the validation [here](https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/mastr.py#L303) and run `db.to_csv("locations_extended")`, this works out.

### Problem 2: CSV encoding

But, haha, the resulting CSV has a wrong encoding (utf-16le):
```console
luke@skywalker:~/.open-MaStR/data$ file -i dataversion-2022-11-17/bnetza_mastr_locations_extended_raw.csv 
dataversion-2022-11-17/bnetza_mastr_locations_extended_raw.csv: application/csv; charset=utf-16le
```

But you explicitly defined the charset:

https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/soap_api/download.py#L411-L428

That might be a different issue?

**Comment by nesnoj** [comment]
Update on the encoding: it seems that only the additional tables have utf-16le:
```console
darth@vader:~/.open-MaStR/data$ for f in dataversion-2022-11-17_OLD/*.csv;do file -i $f;done
dataversion-2022-11-17_OLD/bnetza_mastr_balancing_area_raw.csv: application/csv; charset=utf-16le
dataversion-2022-11-17_OLD/bnetza_mastr_biomass_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_combustion_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_electricity_consumer_raw.csv: application/csv; charset=utf-16le
dataversion-2022-11-17_OLD/bnetza_mastr_gsgk_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_hydro_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_nuclear_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_solar_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_storage_raw.csv: application/csv; charset=utf-8
dataversion-2022-11-17_OLD/bnetza_mastr_wind_raw.csv: application/csv; charset=utf-8
```

**Comment by nesnoj** [comment]
> Update on the encoding: it seems that only the additional tables have utf-16le:
> ```console
> darth@vader:~/.open-MaStR/data$ for f in dataversion-2022-11-17_OLD/*.csv;do file -i $f;done
> dataversion-2022-11-17_OLD/bnetza_mastr_balancing_area_raw.csv: application/csv; charset=utf-16le
> dataversion-2022-11-17_OLD/bnetza_mastr_biomass_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_combustion_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_electricity_consumer_raw.csv: application/csv; charset=utf-16le
> dataversion-2022-11-17_OLD/bnetza_mastr_gsgk_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_hydro_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_nuclear_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_solar_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_storage_raw.csv: application/csv; charset=utf-8
> dataversion-2022-11-17_OLD/bnetza_mastr_wind_raw.csv: application/csv; charset=utf-8
> ```

I opened a separate issue for this problem: #385

**Comment by chrwm** [comment]
The problem with the encoding was simply [fba5bb1](https://github.com/OpenEnergyPlatform/open-MaStR/commit/fba5bb1402ecbfcbd1672d1de3922a986a337f21).
The problem exporting the additional tables was that in https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/mastr.py#L316-L320
`table` was drawn from set of data options of [BULK_DATA](https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/utils/constants.py#L2-L19) and comparead to [ADDITIONAL_TABLES](https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/utils/constants.py#L49-L61), in the case that the tables to export were unspecified in `db.to_csv()`. See below, as method='bulk' is [default](https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/mastr.py#L302-L310).
https://github.com/OpenEnergyPlatform/open-MaStR/blob/a826d158d26ae2a789b4b1d2ae8006143203e305/open_mastr/utils/helpers.py#L273-L274
This resulted in `balancing_area` and `electricity_consumer` beeing the only match between BULK_DATA and ADDITIONAL_TABLES. Thus, only the two were added to the list of exported additional tables.

As solution, I wrote a mapping ([64c306e](https://github.com/OpenEnergyPlatform/open-MaStR/commit/64c306ef36c4a10bb7b247963bdff25dcd2169ab)) from the possible `data` parameters for the bulk download to the corresponding tables names in the sql database and a case to handle it ([f478203](https://github.com/OpenEnergyPlatform/open-MaStR/commit/f4782038f6b4e7c874bc48f0eee8c8fae64d6936))

**Comment by nesnoj** [comment]
Thanks @chrwm for these quick fixes!

**Comment by chrwm** [comment]
Fixed by #384
